mongo/options/autoencryptionoptions.go

```diff
@@ -34,8 +34,14 @@ func AutoEncryption() *AutoEncryptionOptions {
 	return &AutoEncryptionOptions{}
 }
 
-// SetKeyVaultClientOptions specifies options for the client used to communicate with the key vault collection. If this is
-// not set, the client used to do encryption will be re-used for key vault communication.
+// SetKeyVaultClientOptions specifies options for the client used to communicate with the key vault collection.
+//
+// If this is set, it is used to create an internal mongo.Client.
+// Otherwise, if the target mongo.Client being configured has an unlimited connection pool size (i.e. maxPoolSize=0),
+// it is reused to interact with the key vault collection.
+// Otherwise, a separate internal mongo.Client is used (and created if necessary). The internal mongo.Client is
```

Comment:
In what case will it not be created?

Reply:
It will not be created when the target.Client has a maxPoolSize=0.

I tried to clarify this a bit, to also note that the internal client may be shared for metadata operations.

Reply:
The previous line says

> Otherwise, if the target mongo.Client being configured has an unlimited connection pool size (i.e. maxPoolSize=0)...

Given that this line also starts with "Otherwise", we're saying that the condition from the previous line hasn't been met, so the target Client has a non-zero maxPoolSize and the internal Client will always be created in this sceanrio.

Reply:
Yes, an internal `mongo.Client` is always needed in this scenario. What I'd like to convey is that if an internal `mongo.Client` is needed for both key vault and auto encryption, only one is created. Though I'm open to suggestions for clarifying it, since I feel like this is pretty dense.

Reply:
Ah, I see what you're trying to convey. I agree this documentation is a little dense, but I think it's fine for now given that this isn't a feature a beginner user will be looking at.